src/main/scala/com/lightbend/kafka/scala/streams/KGroupedStreamS.scala

```diff
@@ -18,53 +17,18 @@ import FunctionConversions._
  */
 class KGroupedStreamS[K, V](inner: KGroupedStream[K, V]) {
 
-  def count(): KTableS[K, Long] = {
-    val c: KTableS[K, java.lang.Long] = inner.count()
-    c.mapValues[Long](Long2long _)
-  }
-
-  def count(store: String, keySerde: Option[Serde[K]] = None): KTableS[K, Long] = {
-    val materialized = keySerde.foldLeft(Materialized.as[K, java.lang.Long, KeyValueStore[Bytes, Array[Byte]]](store))((m,serde)=> m.withKeySerde(serde))
-
-    val c: KTableS[K, java.lang.Long] = inner.count(materialized)
-    c.mapValues[Long](Long2long _)
-  }
-
-  def reduce(reducer: (V, V) => V): KTableS[K, V] = {
-    inner.reduce((v1, v2) => reducer(v1, v2))
-  }
-
-  def reduce(reducer: (V, V) => V,
-    materialized: Materialized[K, V, KeyValueStore[Bytes, Array[Byte]]]): KTableS[K, V] = {
+  def count()(implicit materialized: Materialized[K, Long, KeyValueStore[Bytes, Array[Byte]]]): KTableS[K, Long] =
+    inner.count(materialized.asInstanceOf[Materialized[K, java.lang.Long, KeyValueStore[Bytes, Array[Byte]]]]).asInstanceOf[KTable[K, Long]]
```

Comment:
This will not work. count() that does not take a `Materialized` *should not* materialize to any store.

Reply:
That's what I was thinking, it's the same as `mapValues`.
I will rollback this one but I think we should keep using the `Materialized` instead of the store name.

Reply:
That's what I was thinking, it's the same as `mapValues`.
I will rollback this one but I think we should keep using the `Materialized` instead of the store name.